db/users.go

```diff
@@ -26,23 +26,25 @@ type PrincipalConfig struct {
 // Check if the password in this PrincipalConfig is valid.  Only allow
 // empty passwords if allowEmptyPass is true.
 func (p PrincipalConfig) IsPasswordValid(allowEmptyPass bool) (isValid bool, reason string) {
-
 	// if it's an anon user, they should not have a password
-	if p.Name == nil && p.Password != nil {
-		return false, "Anonymous users should not have a password"
+	if p.Name == nil {
+		if p.Password != nil {
+			return false, "Anonymous users should not have a password"
+		} else {
+			return true, ""
+		}
 	}
 
 	if allowEmptyPass {
```

Comment:
The previous implementation would allow otherwise invalid passwords, 1 or 2 characters, if allowPass was the. The comment '// allow any password, skip validation' was also there so I left this behaviour but I did ponder it for a while. We can change the logic only allow truly empty passwords. 

P.S. I have reverted the 'dbc.AllowEmptyPassword || allowReplace' to isPasswordValid so this logic should behave as before.